proto/pulumi/provider.proto

```diff
@@ -751,76 +764,160 @@ message DeleteRequest {
     string type = 7;
 }
 
+// `ConstructRequest` is the type of requests sent as part of a [](pulumirpc.ResourceProvider.Construct) call. A
```

Comment:
I'd have a lot of stupid questions here, I guess some intro documentation on components and what's expected from the provider vs the resource would be really appreciated for me to level up.